planning/grpc/server/src/chronicles.rs

```diff
@@ -895,13 +895,14 @@ impl<'a> ChronicleFactory<'a> {
     }
 
     fn set_cost(&mut self, cost: &Expression) -> Result<(), Error> {
-        ensure!(kind(cost)? == ExpressionKind::Constant);
-        ensure!(cost.r#type == "up:integer");
-        let cost = match cost.atom.as_ref().unwrap().content.as_ref().unwrap() {
-            Content::Int(i) => *i as IntCst,
-            _ => bail!("Unexpected cost type."),
+        let value = self.reify(cost, None)?;
+        match value {
+            Atom::Int(i) => {
+                self.chronicle.cost = Some(i);
+                self.chronicle.constraints.push(Constraint::leq(IAtom::ZERO, i));
```

Comment:
This seems incorrect.
Previously, we would disallow user to use negative cost and warn them when encoding the problem.
Here you allow them to do this but would make the problem unsolvable (or at least remove some of the expected solution).

We should check whether there is still a reason to disallow negative costs.

Reply:
The constraint has been removed.
The negative cost are now possible for an action.
See cad2fa0abf70341ed6e834d278ae7b4c2ebaffd1.